backend/commons/src/main/java/org/sonarsource/sonarlint/core/commons/SonarLintBlameResult.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SonarLint Core - Commons
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.commons;
+
+import java.nio.file.Path;
+import java.util.Date;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.sonar.scm.git.blame.BlameResult;
+
+public class SonarLintBlameResult {
+
+  private final BlameResult blameResult;
+
+  public SonarLintBlameResult(BlameResult blameResult) {
+    this.blameResult = blameResult;
+  }
+
+  public Optional<Date> getLatestChangeDateForLinesRangeInFile(Path filePath, LinesRange linesRange) {
+    var fileBlameByPath = blameResult.getFileBlameByPath();
+    var blameForFile = fileBlameByPath.get(filePath.toString());
+    if (blameForFile == null || linesRange.getStartLine() >= blameForFile.lines()) {
+      return Optional.empty();
+    }
+
+    var latestDate = blameForFile.getCommitDates()[linesRange.getStartLine()];
+
+    for (var i = linesRange.getStartLine() + 1; i <= linesRange.getEndLine() && i < blameForFile.lines(); i++) {
+      var dateForLine = blameForFile.getCommitDates()[i];
+      if (isLineRangeModified(dateForLine, latestDate)) {
+        return Optional.empty();
+      }
+      latestDate = latestDate.after(dateForLine) ? latestDate : dateForLine;
+    }
+
+    return Optional.ofNullable(latestDate);
+  }
+
+  private static boolean isLineRangeModified(@Nullable Date dateForLine, @Nullable Date latestDate) {
+    return dateForLine == null || latestDate == null;
+  }
```

Comment:
```suggestion
  private static boolean isLineModified(@Nullable Date dateForLine) {
    return dateForLine == null;
  }
```
I think this method should be responsible for only checking whether the line is modified. Checking the nullability of the latest date seems like a duplication.

Reply:
In the beginning, the `latestDate` is set from the first line of the line range (like this `latestDate = blameForFile.getCommitDates()[linesRange.getStartLine()];`). 
There might be a case when the first line of the line range was modified (i.e. the `latestDate` is `null`), and the other lines are not modified (i.e. the `dateForLine` will not be `null`). That's the reason the method checks both the `dateForLine` and `latestDate` for nullability.

Reply:
Now I see your next comments :-), I can address this as suggested

Reply:
Addressed in [179fcba](https://github.com/SonarSource/sonarlint-core/pull/949/commits/179fcbab50d500abff10bd003bc56d684f5e4102)